brainglobe_atlasapi/atlas_generation/atlas_scripts/cavefish_atlas.py

```diff
@@ -0,0 +1,233 @@
+    ATLAS_PACKAGER = "Robert Kozol, kozolrobert@gmail.com"
+    ADDITIONAL_METADATA = {}
+
+    # setup folder for downloading
+
+    download_dir_path = working_dir / "downloads"
+    download_dir_path.mkdir(exist_ok=True)
+    atlas_path = download_dir_path / "atlas_files"
+
+    # download atlas files
+    utils.check_internet_connection()
+    pooch.retrieve(
+        ATLAS_FILE_URL,
+        known_hash="ee496465f1f55368aa1969180eff91863b800f57ea8dc5e9e3ab592fc480340e",
+        processor=pooch.Unzip(extract_dir=atlas_path),
+        progressbar=True,
+    )
+
+    structures_file = atlas_path / "asty_atlas/SPF2_25_Region_atlas_list.csv"
+    annotations_file = (
+        atlas_path / "asty_atlas/SPF2_regions_SP2c_1iWarp_25.tif"
+    )
+    reference_file = atlas_path / "asty_atlas/SPF2_terk_ref.tif"
+    meshes_dir_path = atlas_path / "asty_atlas/meshes"
+
+    # additional references (not in remote):
+    reference_cartpt = atlas_path / "asty_atlas/SPF2_cartpt_ref.tif"
+
+    try:
+        os.mkdir(meshes_dir_path)
+    except FileExistsError:
+        "mesh folder already exists"
+
+    # create dictionaries
+    print("Creating structure tree")
+    with open(
+        structures_file, mode="r", encoding="utf-8-sig"
+    ) as cavefish_file:
+        cavefish_dict_reader = csv.DictReader(cavefish_file)
+
+        # empty list to populate with dictionaries
+        hierarchy = []
+
+        # parse through csv file and populate hierarchy list
+        for row in cavefish_dict_reader:
+            hierarchy.append(row)
+
+    # make string to int and list of int conversions in
+    # 'id', 'structure_id_path', and 'rgb_triplet' key values
+    for i in range(0, len(hierarchy)):
+        hierarchy[i]["id"] = int(hierarchy[i]["id"])
+    for j in range(0, len(hierarchy)):
+        hierarchy[j]["structure_id_path"] = list(
+            map(int, hierarchy[j]["structure_id_path"].split("/"))
+        )
+    for k in range(0, len(hierarchy)):
+        try:
+            hierarchy[k]["rgb_triplet"] = list(
+                map(int, hierarchy[k]["rgb_triplet"].split("/"))
+            )
+        except ValueError:
+            hierarchy[k]["rgb_triplet"] = [255, 255, 255]
+
+    # remove clear label (id 0) from hierarchy.
+    # ITK-Snap uses this to label unlabeled areas,
+    # but this convention interferes with the root mask generation
+    # and is unnecessary for this application
+    hierarchy.remove(hierarchy[1])
+
+    # Set root mesh to white
+    hierarchy[0]["rgb_triplet"] = [255, 255, 255]
+
+    # use tifffile to read annotated file
+    annotated_volume = tifffile.imread(annotations_file).astype(np.uint8)
+    reference_volume = tifffile.imread(reference_file)
+
+    # segmentation is isotropic 2x2x2, while reference images are ~2x0.5x0.5.
+    # Downsample them to 2x2x2um
+    reference_volume = zoom(
+        reference_volume, (1, 1.0 / 3.995, 1.0 / 3.995), order=0
+    )  # 1/3.995 is a fudge to match annotation dimensions
```

Comment:
I'd like to get this clarified so if we're going to fudge something, we know exactly why.

Reply:
We know don't downsampled the data on our side, but use 2um data from the author.